sysboundary/sysboundarycondition.h

```diff
@@ -143,6 +143,10 @@ namespace SBC {
             cuint& component
          );
         
+         virtual void mapCellPotentialAndGetEXBDrift(
+            std::array<Real, CellParams::N_SPATIAL_CELL_PARAMS>& cellParams
+         )=0;
+
```

Comment:
Hmm, I'm not particularly fond of this change, since it now requires every sysboundarycondition to implement a dummy version of this same function.

And indeed, you implemented it as an error message in all of them except ionosphere, which in turn warns if they have even been called. At the same time, as part of the dAMR merge, the sysboundary class hierarchy has separated out OuterBoundaryCondition, so why not the following:
- Make all the outer boundary conditions inherit from OuterBoundaryCondition, if that isn't already the case
- Implement this "warning, I have been called!" implementation as part of OuterBoundaryCondition
- Let only ionosphere and conductingsphere even hassle with it (the latter returning a simple zero everywhere)

Reply:
Except I need one in the parent class as I am calling it with
```
getObjectWrapper().sysBoundaryContainer.getSysBoundary(sysboundarytype::IONOSPHERE)->mapCellPotentialAndGetEXBDrift(spatial_cell->parameters);
```
in `vlasovsolver/cpu_acc_transform.cpp` so I don't think your suggestion makes much of a difference?

Reply:
> Formal nitpicking: when I was thrown into this object-oriented system boundary business I was told to have one class per file or pair of source+header file. But thanks for the heads-up, this indeed would have saved me a bit of pasting to some of the files. :)

Indeed, you are right. I think it would be good style to move the outerBoundaryCondition into a file of its own. But that shouldn't happen in this pull request. :)

Reply:
OK I think I got it compiling. BRB